### PR TITLE
Updates to Staff and Guest Badges for 2023

### PR DESCRIPTION
**Pull request by kentpmckinney**
**Changes**
- Updates the design and layout of staff and guest badges

**Testing**
- Within the app, navigate to Utility > Export Badges and follow onscreen instructions

### CODE REVIEW

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBack.java [hunk 1]
```diff
@@ -15,15 +15,8 @@ public byte[] createBadge(StaffBadgeDTO staff) {
         drawPositionsStripe(b, staff);
         drawLargeName(b, staff);
         drawPronouns(b, staff);
-        drawBadgeAgeImage(b, staff);
         drawBadgeNumber(b, staff);
-        drawBadgeImage(b, staff);
```
Comment (belac9001) [inline src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBack.java:20]: mostly so i can understand the changes, but do we not draw badge image anymore?

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBase.java [hunk 2]
```diff
@@ -20,118 +18,112 @@ public BadgeCreatorStaffBase(Font boldFont, Font plainFont) {
...
-        if (staff.getPositions() != null) {
-            Rectangle textBounds = new Rectangle(54, 290, 176, 1410);
-//            b.fillRect(textBounds, Color.CYAN);
-            b.drawRotatedCenteredStrings(staff.getPositions(), textBounds, boldFont, fgColor, true);
+        String[] staffPositions = staff.getPositions();
+
+        if (staffPositions != null) {
+            Rectangle textBounds = null;
+
+            if (staffPositions.length == 2) {
```
Comment (kentpmckinney) [inline src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBase.java:41]: clean up

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBase.java [hunk 3]
```diff
@@ -20,118 +18,112 @@ public BadgeCreatorStaffBase(Font boldFont, Font plainFont) {
...
-                b.drawStretchedCenteredString(names[0], line1, boldFont, fgColor, 1);
-            } else if (names.length == 2) {
-                line1 = new Rectangle(270, 1370, 880, 230);
-                line2 = new Rectangle(270, 1530, 880, 160);
-                b.drawStretchedCenteredString(names[0], line1, boldFont, fgColor, 1);
-                b.drawStretchedCenteredString(names[1], line2, boldFont, fgColor, 1);
-            }
+
+        if (names.length == 1) {
+            line1 = new Rectangle(80, 1495, 1020, 200);
```
Comment (kentpmckinney) [inline src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBase.java:83]: clean up

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBase.java [hunk 4]
```diff
@@ -20,118 +18,112 @@ public BadgeCreatorStaffBase(Font boldFont, Font plainFont) {
...
 
     void drawBadgeNumber(BadgeImage b, StaffBadgeDTO staff) {
-        Color fgColor = positionForeground(staff.getDepartmentBackgroundColor());
-        if (staff.getDepartmentBackgroundColor() != null && !staff.getDepartmentBackgroundColor().isBlank()) {
-            Color bgColor = Color.decode(staff.getDepartmentBackgroundColor());
-            fgColor = BadgeImage.getInverseColor(bgColor);
+        Color bgColor = null;
+        if (staff.getDepartment() == null) {
+            // No department == Guest of Honor
+            bgColor = Color.BLACK;
```
Comment (kentpmckinney) [inline src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeCreatorStaffBase.java:118]: check this

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java [hunk 5]
```diff
@@ -39,27 +40,66 @@ public class BadgeImage {
...
     void drawImage(Image image, Rectangle area) {
         if (image == null) return;
+        g2.addRenderingHints(new RenderingHints(RenderingHints.KEY_RENDERING, RenderingHints.VALUE_RENDER_QUALITY));
         g2.drawImage(image, area.x, area.y, area.width, area.height, null);
     }
 
-    void drawStretchedImage(Image image, Rectangle area) {
+    void drawStretchedImage(Image image, Rectangle boundingArea) {
         if (image == null) return;
-        double imageWidth = image.getWidth(null);
```
Comment (kentpmckinney) [inline src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java:47]: revert to original

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java [hunk 6]
```diff
@@ -39,27 +40,66 @@ public class BadgeImage {
...
+        }
+    
+        // then check if we need to scale even with the new height
+        if (new_height > bound_height) {
+            //scale height to fit instead
+            new_height = bound_height;
+            //scale width to maintain aspect ratio
+            // The '* 125 / 100' part is an embarrassing hack to make scaled images 25% wider
+            // since I couldn't figure out what was shrinking them by 25% with the amount of time I had
+            new_width = (new_height * original_width * 125 / 100) / original_height;
```
Comment (kentpmckinney) [inline src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java:99]: hack

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java (hunk 5 above)
Comment (kentpmckinney) [reply src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java:47]: Created issue https://github.com/kumoricon/registration/issues/161 and copied the original code to that issue, leaving it up to whoever resolves that issue what code they want to use here.

File: src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java (hunk 6 above)
Comment (kentpmckinney) [reply src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java:99]: Created issue https://github.com/kumoricon/registration/issues/161 to resolve the root cause of this

**Pull request merged**